editor/state.js

```diff
@@ -91,13 +96,28 @@ export const blocks = combineUndoableReducers( {
 					...state.slice( index + 2 )
 				];
 
+			case 'REPLACE_BLOCKS':
+				if ( ! action.blocks ) {
+					return state;
+				}
+				index = state.indexOf( action.uids[ 0 ] );
+				return state.reduce( ( memo, uid ) => {
+					if ( uid === action.uids[ 0 ] ) {
```

Comment:
I'm not assuming they are in a consecutive order, I'm just inserting the new blocks in a consecutive order at the place of the first old block. 

But I'm not sure I'll keep this implementation for now. My preference would be to transform one block into another block (1-1) transformation but we'll see with the discussion.